server/src/main/java/com/broadcom/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -190,6 +194,10 @@ public Class visitProgramUnit(ProgramUnitContext ctx) {
   public Class visitProcedureSection(ProcedureSectionContext ctx) {
     throwWarning(ctx.getStart());
     outlineTreeBuilder.addNode(ctx.getStart().getText(), NodeType.PROCEDURE_SECTION, ctx);
+
+    String name = ctx.getStart().getText().toUpperCase();
+    getLocality(ctx.getStart())
+        .ifPresent(locality -> paragraphDefinitionLocalities.put(name, locality));
```

Comment:
A section is not a paragraph, it should not be consumed here

Reply:
Discussed in the chat. Closed.